engine/src/test/java/com/sandboni/core/engine/sta/operation/AllReachableEdgesOperationTest.java

```diff
@@ -35,29 +35,54 @@ public void getAllReachableEdges() {
         assertEquals(caller, edge1.getTarget());
 
         Edge edge2 = iterator.next();
-        assertEquals(CUCUMBER_TEST, edge2.getLinkType());
+        assertEquals(METHOD_CALL, edge2.getLinkType());
         assertEquals(modified, edge2.getSource());
-        assertEquals(affectedCucumberTest, edge2.getTarget());
+        assertEquals(caller_ExplicitReflection, edge2.getTarget());
 
         Edge edge3 = iterator.next();
         assertEquals(METHOD_CALL, edge3.getLinkType());
-        assertEquals(caller, edge3.getSource());
-        assertEquals(callerTest, edge3.getTarget());
+        assertEquals(modified, edge3.getSource());
+        assertEquals(callerTest_ExplicitReflection, edge3.getTarget());
 
         Edge edge4 = iterator.next();
-        assertEquals(ENTRY_POINT, edge4.getLinkType());
-        assertEquals(callerTest, edge4.getSource());
-        assertEquals(START_VERTEX, edge4.getTarget());
+        assertEquals(CUCUMBER_TEST, edge4.getLinkType());
```

Comment:
I don't get what's the reason for these changes, was this just due reordering the lines? Try to avoid it because:
1. Complicates the review process.
2. These tests where the links are asserted are critical for the core, if there's no valid reason to change them, is better to create a new test and validate only new links.

Reply:
I've added reflection scenarios, which added reachable edges. 
since this TreeSet is ordered, I had to insert my edges in the right locations of the iteration.

Reply:
I think would be better to create a new test instead of modifying the existing tests. Tests related to the graph are our safety net to make sure that current things are not broken with new features.